src/main.rs

```diff
@@ -9,7 +9,11 @@ use std::cell::RefCell;
 use std::error::Error;
 use std::fmt;
 use std::str::FromStr;
-use tonic_lnd::{Client, ConnectError};
+use tonic_lnd::{tonic::Code, Client, ConnectError};
+
+const ONION_FEATURE_BIT: u32 = 39;
```

Comment:
nitpick: name `onion_messages_optional` (or something of the sort) to align with naming in the spec.